setup.py

```diff
@@ -28,6 +28,14 @@
     'lint': lint_deps,
 }
 
+commithash = subprocess.check_output("git rev-parse HEAD".split())
+commithash = commithash.decode('utf-8').strip()
+
+hash_file_rel_path = os.path.join('vyper', 'git_version.txt')
```

Comment:
This was my original approach but I feel kind of weird because it causes the build to leave files in the source directory.

Reply:
It could also overwrite some programmer file named `'git_version.txt`'. I think the only way it's OK to use the source directory if a) we make sure the file is cleaned up afterward and b) there is a check that the file doesn't already exist. Tempfile handles even pathological cases (OOM reaper, power disconnect), but I guess it would be OK to just use a `try: finally` block.

Reply:
Couldn't we just add it to `.gitignore`?

Reply:
Already added to gitignore, and it's no concern really for programmers (because the only folks that care about are vyper devs - aka. us). If you install it from git checkout you get the eight hash.

Reply:
Also build _always_ leaves a bunch of files in anycase.

Reply:
Also should check that this approach works from `pip install vyper` and `pip install git+https://github.com/ethereum/vyper`, as well as `pip install .` and `pip install ../vyper_repo`

Reply:
Oh the `.gitignore` commit went here `https://github.com/ethereum/vyper/pull/1537`, I am just gonna leave it as is.

Reply:
We can check the first, as the requires a release cut... which I check every release cut in any case.

Reply:
But the second type of ones work fine: 
```
(vyper-pip-test) jacques at deadlock in ~
$ vyper --version
0.1.0b10
GIT:f587d2cae2e6e8c65017e290367c077515b2399c

```

Reply:
The build process leaves files in build directories, not source directories.

Reply:
Oooh I see, we could check to put it in build/ ?

Reply:
Yeah something like that, `build/` is managed by setuptools so it's a bit brittle because we run the risk of overwriting a file that setuptools creates, but we should be pretty safe if we call it something like `build/__vyper_git_version__.txt`.

Reply:
```suggestion
hash_file_rel_path = os.path.join('vyper', '__vyper_git_version__.txt')
```